### PR TITLE
os/arch/arm/src/armv7-a: Modify assert handler for SMP case

### PR DESCRIPTION
**Pull request by kishore-sn**
If an assert happens when SMP is handled, we need to do the following:
- Enter critical section
- If crash is in kernel, then stop all other cores
- Perform assert handling
- Leave critical section

### CODE REVIEW

**Review by samsung-singh — APPROVED** [review]
assert handler is modified in the case of SMP, this change is good to merge.

File: os/arch/arm/src/armv7-a/arm_assert.c
Comment (sunghan-chang) [inline os/arch/arm/src/armv7-a/arm_assert.c:570]: In this place, is this working? As far as I know, if we can this at middle of a function, it does not give a proper address. If we want to move this here, then I think we should use __builtin_..... instead of ARCH_GET....

File: os/arch/arm/src/armv7-a/arm_assert.c
Comment (sunghan-chang) [inline os/arch/arm/src/armv7-a/arm_assert.c:573]: How about leaving some comment to explain below?

**Comment by sunghan-chang** [comment]
@kishore-sn There is a build break with flat_apps.
```
make[2]: Leaving directory '/*******/os/board/rtl8730e/src'
LD: tinyara
/*******/os/../build/output/libraries/libarch.a(arm_assert.o): In function `up_assert':
/*******/os/arch/arm/src/armv7-a/arm_assert.c:574: undefined reference to `is_kernel_space'
Makefile:207: recipe for target '../build/output/bin/tinyara.axf' failed
make[1]: *** [../build/output/bin/tinyara.axf] Error 1
```

File: os/arch/arm/src/armv7-a/arm_assert.c
Comment (ritesh55555) [inline os/arch/arm/src/armv7-a/arm_assert.c:574]: there is a build break in flat because this is only for loadable config, please check

File: os/arch/arm/src/armv7-a/arm_assert.c
Comment (edwakuwaku) [inline os/arch/arm/src/armv7-a/arm_assert.c:576]: let's use meaningful variable, maybe `int cpu`

File: os/arch/arm/src/armv7-a/arm_assert.c
Comment (kishore-sn) [reply os/arch/arm/src/armv7-a/arm_assert.c:570]: Right.. i reverted it. Thank you.

File: os/arch/arm/src/armv7-a/arm_assert.c
Comment (kishore-sn) [reply os/arch/arm/src/armv7-a/arm_assert.c:573]: Done. Thanks.

File: os/arch/arm/src/armv7-a/arm_assert.c
Comment (sunghan-chang) [inline os/arch/arm/src/armv7-a/arm_assert.c:557]: Let's leave this.

Comment (sunghan-chang) [reply os/arch/arm/src/armv7-a/arm_assert.c:557]: @r-prabu After this, let's merge this PR without my approval.

**Pull request merged**
